paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +688,62 @@ def validate(
             return []
 
 
+def generate_vitess_instance_config(
+    instance_config: VitessDeploymentConfigDict,
+) -> VitessDeploymentConfigDict:
+    # Generate vitess instance config from yelpsoa config
+    zk_address = instance_config.get("zk_address")
+    cells = instance_config.get("cells")
+    keyspaces = instance_config.get("keyspaces")
+    region = instance_config.get("region")
+    vtgate_resources = instance_config.get("vtgate_resources")
+    vtadmin_resources = instance_config.get("vtadmin_resources")
+    vtctld_resources = instance_config.get("vtctld_resources")
+    deploy_group = instance_config.get("deploy_group")
+
+    # healthcheck cmd and mode are required to be in config_dict for LongRunningServiceConfig
+    healthcheck_cmd = instance_config.get("healthcheck_cmd")
+    healthcheck_mode = instance_config.get("healthcheck_mode")
+
+    vitess_images = load_system_paasta_config().get_vitess_images()
+
+    vitess_instance_config: dict = {
+        "zk_address": zk_address,
+        "cells": cells,
+        "region": region,
+        "vtgate_resources": vtgate_resources,
+        "vtadmin_resources": vtadmin_resources,
+        "vtctld_resources": vtctld_resources,
+        "keyspaces": keyspaces,
+        "images": vitess_images,
+        "updateStrategy": {"type": "Immediate"},
+        "healthcheck_cmd": healthcheck_cmd,
+        "healthcheck_mode": healthcheck_mode,
+        "deploy_group": deploy_group,
+    }
+    return VitessDeploymentConfigDict(vitess_instance_config)  # type: ignore
+
+
 def load_vitess_instance_config(
     service: str,
     instance: str,
     cluster: str,
+    instance_type: str = "vitesscluster",
```

Comment:
fwiw: it should be fine to keep hardcoding vitesscluster - vitess won't need to migrate to EKS since it'll only ever exist on eks clusters so i don't see the instance type name changing in the foreseeable future :)

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/48ebe609e02377336248dcbb9bfed29a0d6213ae